pkg/controller/s3/bucket/replicationConfig.go

```diff
@@ -214,3 +212,153 @@ func sortReplicationRules(rules []awss3.ReplicationRule) {
 		}
 	}
 }
+
+// LateInitialize does nothing because the resource might have been deleted by
+// the user.
+func (in *ReplicationConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketReplicationRequest(&awss3.GetBucketReplicationInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		return awsclient.Wrap(resource.Ignore(s3.ReplicationConfigurationNotFound, err), replicationGetFailed)
+	}
+
+	if external.GetBucketReplicationOutput == nil || external.ReplicationConfiguration == nil || len(external.ReplicationConfiguration.Rules) == 0 {
+		return nil
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+	fp := &bucket.Spec.ForProvider
+	if fp.ReplicationConfiguration == nil || len(fp.ReplicationConfiguration.Rules) == 0 {
+		// We need the configuration to exist so we can initialize
+		fp.ReplicationConfiguration = &v1beta1.ReplicationConfiguration{}
+	} else {
+		return nil
+	}
+
+	createReplicationRulesFromExternal(external.ReplicationConfiguration, fp.ReplicationConfiguration)
+	return nil
+}
+
+func createReplicationRulesFromExternal(external *awss3.ReplicationConfiguration, config *v1beta1.ReplicationConfiguration) { // nolint:gocyclo
+	config.Role = awsclient.LateInitializeStringPtr(config.Role, external.Role)
+	if config.Rules == nil {
+		config.Rules = make([]v1beta1.ReplicationRule, 0)
+	}
+
+	for i, rule := range external.Rules {
+		if i == len(config.Rules) {
+			config.Rules = append(config.Rules, v1beta1.ReplicationRule{})
+		}
+		config.Rules[i] = v1beta1.ReplicationRule{
+			ID:       awsclient.LateInitializeStringPtr(config.Rules[i].ID, rule.ID),
+			Priority: awsclient.LateInitializeInt64Ptr(config.Rules[i].Priority, rule.Priority),
+			Status:   awsclient.LateInitializeString(config.Rules[i].Status, awsclient.String(string(rule.Status))),
+		}
+		if rule.Filter != nil {
+			if config.Rules[i].Filter == nil {
```

Comment:
```suggestion
```

We don't need nil guards or use of LateInitialize* function usage for `config.Rules[i]` objects since we initialize them in the beginning of the function, so it's safe to assign directly.